sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/RowEncoderSuite.scala

```diff
@@ -185,6 +185,20 @@ class RowEncoderSuite extends SparkFunSuite {
     assert(encoder.serializer.head.nullable == false)
   }
 
+  test("RowEncoder should support array as the external type for ArrayType") {
```

Comment:
This is fine for now, but should we build the equivalent of `encodeDecodeTest` for `Row`s? or is this the only special case?